queries/c_sharp/highlights.scm

```diff
@@ -364,6 +364,7 @@
  "readonly"
  "static"
  "volatile"
+ "required"
 ] @storageclass
```

Comment:
I think this should be `@type.qualifier`.

Reply:
this is why I left this PR open. For me there is no obvious distinction between those two highlight groups.

Reply:
In this case, do you guys think we should move `const` & `readonly` from `@storageclass` to `@type.qualifier` as well ?

Reply:
I agree with both of you. `@type.qualifier` sounds better than `@storageclass`

Reply:
I looked into the C# documentation because I didn't yet know of this keyword.
Turns out they classify it the same as the `static` keyword. So making it a `@storageclass` makes the most sense to be consistent.
So I am changing my mind. `@storageclass` makes the most sense, So let's keep it like that for now 😛.
Here are the links for reference: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/required and https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/static.